crates/e2e/tests/e2e/protocol_fee.rs

```diff
@@ -0,0 +1,245 @@
+use {
+    e2e::{
+        setup::{colocation::SolverEngine, *},
+        tx,
+    },
+    ethcontract::prelude::U256,
+    model::{
+        order::{OrderCreation, OrderKind},
+        signature::EcdsaSigningScheme,
+    },
+    secp256k1::SecretKey,
+    shared::ethrpc::Web3,
+    web3::signing::SecretKeyRef,
+};
+
+#[tokio::test]
+#[ignore]
+async fn local_node_price_improvement_fee_sell_order() {
+    run_test(price_improvement_fee_sell_order_test).await;
+}
+
+#[tokio::test]
+#[ignore]
+async fn local_node_price_improvement_fee_sell_order_capped() {
+    run_test(price_improvement_fee_sell_order_capped_test).await;
+}
+
+#[tokio::test]
+#[ignore]
+async fn local_node_price_improvement_fee_buy_order() {
+    run_test(price_improvement_fee_buy_order_test).await;
+}
+
+#[tokio::test]
+#[ignore]
+async fn local_node_price_improvement_fee_buy_order_capped() {
+    run_test(price_improvement_fee_buy_order_capped_test).await;
+}
+
+// Without protocol fee:
+// Expected execution is 10000000000000000000 GNO for
+// 9871415430342266811 DAI, with executed_surplus_fee = 167058994203399 GNO
+//
+// With protocol fee:
+// surplus in buy token = 9871415430342266811 - 5000000000000000000 =
+// 4871415430342266811
+//
+// protocol fee in buy token = 0.3*surplus = 1461424629102680043
+//
+// protocol fee in sell token = 1461424629102680043 / 9871415430342266811 *
+// (10000000000000000000 - 167058994203399) = 1480436341679873337
+//
+// expected executed_surplus_fee is 167058994203399 + 1480436341679873337 =
+// 1480603400674076736
```

Comment:
It's not a great sign that these tests need so much documentation (which will very likely not age well). I understand that the test setup code is so much boilerplate that we want to combine it for all test. I think this is a sign that very specific test cases (that check exact math) should go in the driver, where the setup is ergonomic enough to do it once for each test.

Maybe it would help slightly to move the comments just above the `execute_test` args that determine the case (this is very personal pref, feel free to ignore).

Maybe we can have the values documented